rfd/0001-general-design.md

```diff
@@ -0,0 +1,288 @@
+│      ┌──────┐      │    │            ╲ Cache ╱          │├──────┼────┼──────────┤ │
+│      │ IP 3 │──────┼────┤             ╲     ╱           ││IPADDR│ 5  │*time.Time│ │
+│      └──────┘      │    │              ╲   ╱            │└──────┴─▲──┴──────────┘ │
+│      ┌──────┐      │    │               ╲ ╱             │    ┌────┘               │
+│      │ IP N │──────┼────┘                V              └────┼────────────────────┘
+│      └──────┘      │             ┌──────────────┐            │
+│                    │             │   In Cache   │       ┌─────────┐
+└────────────────────┘             └──────────────┘       │         │
+                                           │              │Increment│
+                                           │              │ counter │
+                                           └─────────────▶│         │
+                                                          │If < than│
+                                                          │threshold│
+                                                          │         │
+                                                          └─────────┘
+```
+
+
+
+### Security Authentication and Credential provision
+
+#### Client Service mTLS Layer
+Proposed mTLS layer consists of the following scheme:
+```
+                                                                    ╔════════════╗
+                                                      ┌─────────────╣ Connection ╠─────────────┐
+                                                      │             ║  manager   ║             │
+                                                      │             ╚════════════╝             │
+                                                      │                                        │
+┌──────────┐                                          │       ┌───────────────┐                │
+│          │                                       ┌──┼──────▶│  Accept Cert  │                │
+│  CACert  │───────┐        ┌────────────┐         │  │       └───────┬───────┘                │
+│          │       │        │            │         │  │               │                        │
+└──────────┘  ┌────┴──┐     │   Client   │  ┌──────┴┐ │       ┌───────▼───────┐                │
+              │Produce│     │    Cert    ├──┤Connect│ │       │  Fetch CN     │                │
+┌──────────┐  │  TLS  ├────▶│            │  └───────┘ │       └───────────────┘                │
+│  Client  │  └────┬──┘     │ CN=Identity│            │               │                        │
+│ Frontend │       │        │            │            │       ┌───────┘                        │
+│  Access  │───────┘        └────────────┘            │       │                                │
+│    key   │                                         ┌┴───────▼───────┐                        │
+└──────────┘                                         │ Lookup Access  │    ┌─────────────────┐ │
```

Comment:
Yes, I was changing the naming for clarity. At least in my head, I was thinking it would bring one :)

But answering the question:
- CN provides identity key
- Identity Key placed in the CN by outer layers of the application importing the library
- Identity Key is the Identity of the Cluster Pool record (can be UUID of customer pool) or token derived from the customer pool

Let me know if my assumption needs to be corrected for this area, as I prefer to use CN for this type of authentication and might be unaware if other fields are better for this particular goal.

Reply:
Using the Common Name/Subject seems like a good strategy here and does match a majority of x509 client cert auth implementations in the wild.  I think a little additional explanation on how the mapping/authorization will work here would be useful - how will access be granted to specific identities, are you pursuing a RBAC model or something different (e.g users are directly assigned access to certain applications).

Reply:
Thanks, Noah; as I said above, I updated the document but didn't request the review. But let me see if I need to include additional information after your comment above.

There are certainly new Diagrams there showing how it should work, but I will try to re-check if additional details can be included.